cloudmarker/clouds/gcpcloud.py

```diff
@@ -58,6 +56,16 @@ def __init__(self, key_file_path, zone):
         # https://developers.google.com/api-client-library/python/apis/
         self._compute_resource = self._get_resource('compute', 'v1')
 
+        # Fetching all available zones for the current project.
+        get_zones = self._compute_resource.zones()
+        list_zones = get_zones.list(project=self._project_name)
+        project_zones = list_zones.execute()
```

Comment:
The zone iteration code should move to `read()` method. The `__init__()` method should be fairly lightweight.